synapse_data_warehouse/synapse/dynamic_tables/V2.21.0__verificationsubmission_latest.sql

```diff
@@ -0,0 +1,38 @@
+use schema {{database_name}}.synapse; --noqa: JJ01,PRS,TMP,CP01
+
+CREATE OR REPLACE DYNAMIC TABLE VERIFICATIONSUBMISSION_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+    WITH latest_rows AS (
+        SELECT
+            id AS latest_id,
+            MAX(snapshot_timestamp) AS latest_timestamp
+        FROM
+            verificationsubmissionsnapshots
+        GROUP BY
+            latest_id
+    ),
+    latest_unique_rows AS (
```

Comment:
I chose to work with `ROW_NUMBER` instead of `DISTINCT` because the `DISTINCT` solution was leading to a duplicate row where there was a discrepancy in `CREATED_ON`. The timestamp for this column was off my milliseconds and created this duplicate:

<img width="223" alt="image" src="https://github.com/user-attachments/assets/f18e8709-85a9-4dd6-8ada-0542301d579d">
 
My solution strictly enforces no-duplicates, but at the cost of dropping any duplicate silently. I'm opening this up for discussion in case we want to take a more cautious approach.

Reply:
Could you speak to why you didn't choose to partition by ID and order by snapshot date to get the ranking?

Reply:
This was a typo and does not reflect the final solution I have written on my SQL worksheet. Something weird must have happened in the copy/paste process. Thanks for pointing it out!

Reply:
A good indication of why your clone solution is a good future solution to work towards!